Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -3067,7 +3080,7 @@ def _inject_locals(self, node, func_name):
                           % len(node.args))
                 if len(node.args) > 0:
                     return node # nothing to do
-            return ExprNodes.LocalsExprNode(pos, self.current_scope_node(), lenv)
+            return ExprNodes.LocalsExprNode(pos, def_node, lenv)
```

Comment:
I think `def_node` might not have been assigned here in some cases. Feel free to always assign it outside of the condition above.